pkg/serializer/encode.go

```diff
@@ -98,6 +117,117 @@ func (e *encoder) EncodeForGroupVersion(fw FrameWriter, obj runtime.Object, gv s
 		encoder = serializerInfo.PrettySerializer
 	}
 
+	// Get a version-specific encoder for the specified groupversion
+	versionEncoder := e.codecs.EncoderForVersion(encoder, gv)
+
+	// If the user asked to preserve comments and the contenttype is YAML, try to preserve comments
+	// If preserving comments was asked for, but the content type isn't YAML, continue with normal operation
+	if *e.opts.PreserveComments && fw.ContentType() == ContentTypeYAML {
+		err, ok := e.tryToPreserveComments(versionEncoder, fw, obj)
```

Comment:
err -> result, for clarity. We want to conditionally return `nil` from this as well.

Reply:
rewrote structure